### ISSUE TITLE
Building OCE as a static library fails on VS 2013

### ISSUE THREAD

**Issue by Stinkfist0** [tpaviot/oce#564]
Building OCE as a static library using VS 2013 (Update 4) and Windows SDK 8.1 fails due to unfortunate precompiler definitions leaking from `windows.h` and the files it includes subsequently.

As a quick fix one can do the following additions:
in inc\Visual3d_Layer.hxx:
```c
#ifdef DrawText
#undef DrawText
#endif
```

in src\AIS\AIS_ConcentricRelation.cxx:
```c
#ifdef rad1
#undef rad1
#endif
#ifdef rad2
#undef rad2
#endif
```

in src\Voxel\Voxel_Writer.cxx:
```c
#ifdef small
#undef small
#endif
```

I think a good proper fix could be having a file similar to this:
https://github.com/juj/MathGeoLib/blob/master/src/Math/InclWindows.h
In this file both "best practive" and necessary precautions could be taken (defining WIN32_MEAN_AND_LEAN, NOMINMAX, NOGDI, etc.) before including `windows.h` and undefining possible code breaking definitions (DrawText. small, arg1 and arg2 in this case). right after the windows.h inclusion. This file would be used to safely include windows.h throughout the whole OCE codebase.

**Comment by QbProg** [comment]
Thanks I'm fixing it this night in the branch qb/improvements
When I'm ready I'll ask you to kindly test it and give a feedback :)

2015-07-20 22:04 GMT+02:00 Ali Kämäräinen <notifications@github.com>:

**Comment by Stinkfist0** [comment]
Great!

**Comment by Stinkfist0** [comment]
FYI, these same errors occur also on VS 2015. VS 2015 dynamic lib build also has some problems, I will provide a PR to fix these.

**Comment by QbProg** [comment]
The branch qb/improvements should already contain some fixes, you could
cherry-pick them.

Il giorno mar 18 ago 2015 alle ore 21:20 Ali Kämäräinen <
notifications@github.com> ha scritto:

**Comment by tpaviot** [comment]
@QbProg If you consider some changes could be of interest, could you please rebase qb/improvements and create a pull request ?

**Comment by QbProg** [comment]
The main purpose of that branch was to separate the the bundle from the OCE
CMakeLists.txt and make sure it works in all windows config variants. It's
almost ready I need to finish mingw support which takes forever to build
and merge some commits. Also I would like to integrate samples and enable
vtk, but these could be done in a second PR.
Also, I would like to enable the tcl test suite that doesn't run with the
current approach
Qb

Il giorno mer 19 ago 2015 08:46 Thomas Paviot <notifications@github.com> ha
scritto:

**Comment by Stinkfist0** [comment]
Also one notable warning that can be fixed e.g. by using `#define NOGDI`:
```
oce\src\Resource\Resource_Manager.cxx(42): warning C4005: 'ERROR': macro redefinition
C:\Program Files (x86)\Windows Kits\8.1\Include\um\wingdi.h(114): note: see previous definition of 'ERROR' (compiling source file C:\Dev\GitHub\Tridify\oce\src\Resource\Resource_Manager.cxx)
```

**Comment by QbProg** [comment]
Fixed by PR #575

**Comment by Stinkfist0** [comment]
Great!
